### PR TITLE
Query security parameter from the local node, pass it to all indexers

### PR DESCRIPTION
**Pull request by eyeinsky**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [X] Commit sequence broadly makes sense
    - [X] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [X] Self-reviewed the diff
    - [X] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [X] Reviewer requested

### CODE REVIEW

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs [hunk 1]
```diff
@@ -110,11 +110,11 @@ initialCoordinator indexerCount =
 -- The points should/could provide shared access to the indexers themselves. The result
 -- is a list of points (rather than just one) since it offers more resume possibilities
 -- to the node (in the unlikely case there were some rollbacks during downtime).
-type Worker = Coordinator -> FilePath -> IO [Storable.StorablePoint ScriptTx.ScriptTxHandle]
+type Worker = Int -> Coordinator -> FilePath -> IO [Storable.StorablePoint ScriptTx.ScriptTxHandle]
```
Comment (koslambrou) [inline marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs:113]: Can we use `Word64` as that's the type used by consensus as well?

Also, how about a newtype for this?

File: marconi-chain-index/src/Marconi/ChainIndex/Utils.hs [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
+    eraInMode :: C.EraInMode C.BabbageEra C.CardanoMode
+    eraInMode = C.BabbageEraInCardanoMode
+
+    queryInMode :: C.QueryInMode C.CardanoMode (Either EraMismatch C.GenesisParameters)
+    queryInMode = C.QueryInEra eraInMode
+      $ C.QueryInShelleyBasedEra (C.shelleyBasedEra @C.BabbageEra) C.QueryGenesisParameters
+
+    -- TODO /Really/ handle the error.
+    showError :: Show a => a -> b
+    showError = error . show
```
Comment (koslambrou) [inline marconi-chain-index/src/Marconi/ChainIndex/Utils.hs:39]: Why not handle it in this PR? Do you think the resulting change will be too big?

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs [hunk 3]
```diff
@@ -385,18 +385,19 @@ mintBurnWorker_ bufferSize onInsert Coordinator{_barrier} ch dbPath = do
...
   void $ forkIO loop
   readMVar ix >>= Storable.resumeFromStorage . view Storable.handle
 
 initializeIndexers
-  :: [(Worker, FilePath)]
+  :: Int
+  -> [(Worker, FilePath)]
   -> IO ([ChainPoint], Coordinator)
-initializeIndexers indexers = do
+initializeIndexers securtyParam indexers = do
```
Comment (koslambrou) [inline marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs:398]: Typo

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs [hunk 4]
```diff
@@ -385,18 +385,19 @@ mintBurnWorker_ bufferSize onInsert Coordinator{_barrier} ch dbPath = do
...
   coordinator <- initialCoordinator $ length indexers
-  startingPoints <- mapM (\(ix, fp) -> ix coordinator fp) indexers
+  startingPoints <- mapM (\(ix, fp) -> ix securtyParam coordinator fp) indexers
```
Comment (koslambrou) [inline marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs:400]: Typo

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs (hunk 1 above)
Comment (berewt) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs:113]: I'd avoid a newtype here: worker is really a type alias, I don't see a scenario where we'd like to distinguish `Worker` from its alias  (but I'd be happy to change my mind if you have one), which is the main (unique/) purpose of a newtype.

File: marconi-chain-index/src/Marconi/ChainIndex/Utils.hs (hunk 2 above)
Comment (eyeinsky) [reply marconi-chain-index/src/Marconi/ChainIndex/Utils.hs:39]: Using `error` is what we do elsewhere as well. So as we don't have any specific way to do it then how do we do it?

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs (hunk 1 above)
Comment (eyeinsky) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs:113]: @koslambrou Newtype for the security parameter you mean?

Comment (koslambrou) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs]: Yes :) Newtype for the security param, not the worker.

File: marconi-chain-index/src/Marconi/ChainIndex/Utils.hs (hunk 2 above)
Comment (koslambrou) [reply marconi-chain-index/src/Marconi/ChainIndex/Utils.hs:41]: Alright. Then that means we need a story for error handling

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs (hunk 1 above)
Comment (berewt) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs]: My bad, so yes.

Comment (eyeinsky) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs]: 21830c7d1

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs (hunk 4 above)
Comment (eyeinsky) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs]: 7d440028c

**Review by koslambrou — APPROVED** [review]

**Pull request merged**